sklearn/ensemble/_hist_gradient_boosting/_grower.pyx

```diff
@@ -0,0 +1,67 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: language_level=3
+
+import numpy as np
+cimport numpy as np
+
+from .common cimport node_struct
+from ._predictor cimport TreePredictor
+
+np.import_array()
+
+def _fill_predictor_node_array(TreePredictor predictor_nodes,
```

Comment:
Can you help me understand why we need to redefine this? It seems that the fields of the `nodes` vector are still accessible from withing pure Python, as suggested by the use of `         assert nodes[0]['count'] == n_samples ` in the tests

Reply:
We still need access to the vector to modify it. The `PredictorTree.nodes` is a copy of the vector of structs.